tests/attention/test_mla_functional.py

```
@@ -0,0 +1,371 @@
+    _compare_result(case, result, expected_output, expected_lse)
+
+
+def test_functional_exports_and_legacy_signature_are_stable():
+    assert hasattr(flashinfer.mla, "batch_mla_paged_attention")
+    assert not hasattr(flashinfer.decode, "batch_mla_paged_attention")
+    assert not hasattr(flashinfer, "batch_mla_paged_attention")
+    assert inspect.signature(
+        flashinfer.mla.batch_mla_paged_attention
+    ) == inspect.signature(flashinfer.mla.trtllm_batch_decode_with_kv_cache_mla)
+    namespace = {}
+    exec("from flashinfer.mla import *", namespace)
```

Ruff (0.16.1)

[error] Use of `exec` detected

(S102)